presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -381,6 +382,30 @@ public Connection getConnection(JdbcOutputTableHandle handle)
         return driver.connect(handle.getConnectionUrl(), toProperties(handle.getConnectionProperties()));
     }
 
+    @Override
+    public Statement getStatement(Connection connection)
+            throws SQLException
+    {
+        Statement statement;
+        String dataBaseName = connection.getMetaData().getDatabaseProductName();
+        if (dataBaseName.contains("PostgreSQL")) {
+            connection.setAutoCommit(false);
+            statement = connection.createStatement();
+            statement.setFetchSize(FETCHSIZE);
+        }
+        else if (dataBaseName.contains("MySQL")) {
+            statement = connection.createStatement(java.sql.ResultSet.TYPE_FORWARD_ONLY,
```

Comment:
There is a MySQL specific method to enable streaming. See the linked issue for details.